tedana/utils.py

```diff
@@ -229,9 +233,43 @@ def make_adaptive_mask(data, mask, threshold=1, methods=["dropout"]):
             "Removing voxels from mask."
         )
         adaptive_mask[adaptive_mask < threshold] = 0
+    if isinstance(n_independent_echos, int):
+        # For EPTI sequences, the way we use adaptive mask thresholding fails
+        # because sequential echoes have overlapping information and
+        # there is no clear mapping between the independent sources and the echoes.
+        # Since EPTI has less dropout, it is unclear how often this will cause issues.
+        # To track this, we are flagging voxels that might mark less independent signal.
+        # If such voxels appear often, this would show we might need to alter how the mask is used.
+        # The thresh where there might not be 3 independent sources of data within the good echoes
+        # For n_echos=100 & n_independent_echos=3, threshold_dof=66.
+        # For n_echos=100 & n_independent_echos=4, threshold_dof=50
+        threshold_3dof = np.floor(2 * n_echos / n_independent_echos)
+        n_3dof_voxels = np.sum(np.logical_and(adaptive_mask < threshold_3dof, adaptive_mask >= 1))
+        perc_3dof_voxels = 100 * n_3dof_voxels / np.sum(adaptive_mask >= 1)
+        if perc_3dof_voxels > 0:
+            LGR.warning(
+                f"{n_3dof_voxels} voxels ({np.round(perc_3dof_voxels, decimals=1)}%) have fewer "
+                f"than {np.round(threshold_3dof)} "
+                "good voxels. These voxels will be used in all analyses, "
+                "but might not include 3 independent echo measurements."
+            )
 
+        # There's a separate warning about DOF if it's possible there's a DOF reduction.
+        if n_independent_echos > 3:
+            # The threshold where the loss of good echoes might affect the DOF
+            # For n_echos=100 & n_independent_echos=4, threshold_dof=75
+            threshold_dof = np.floor((n_independent_echos - 1) * n_echos / n_independent_echos)
+            n_dof_voxels = np.sum(
+                np.logical_and(adaptive_mask < threshold_dof, adaptive_mask >= 1)
+            )
+            perc_dof_voxels = 100 * n_dof_voxels / np.sum(adaptive_mask >= 1)
+            LGR.warning(
+                f"{n_dof_voxels} voxels ({np.round(perc_dof_voxels, decimals=1)}%) have fewer"
+                f"than {np.round(threshold_dof)} good voxels."
+                f"The degrees of freedom for fits across echoes will remain {n_independent_echos}"
```

Comment:
```suggestion
                f"{n_dof_voxels} voxels ({np.round(perc_dof_voxels, decimals=1)}%) have fewer "
                f"than {np.round(threshold_dof)} good voxels. "
                f"The degrees of freedom for fits across echoes will remain {n_independent_echos} "
```

There might be other issues, but this definitely the cause of some crashes. There are automatic style changes on commit, and I think it adjusted the lines of this message and got rid of extra spaces. The text read "fewerthan" which correctly flagged an error for the unit test.